js/pokemon.js

```diff
@@ -534,38 +560,52 @@ function checkPokemonLoaded() {
  */
  
 function updateStateAndRefreshUI() {
+
+    //Checks to see if the generation selection has changed
+    if(!_.isEqual(currentGen, newGen)) {
         
-    if(newGen != currentGen) {
-        // The generation has been updated, so highlight the new one
-        document.getElementById('gen' + currentGen).className = document.getElementById('gen' + currentGen).className.replace('current','');
-        document.getElementById('gen' + newGen).className = document.getElementById('gen' + currentGen).className.replace('selected','');
-        document.getElementById('gen' + newGen).className += ' current';
-        currentGen = newGen;
+        //First we remove selected and current from all the gens
+        $('.genSelect').removeClass('selected current');
+        //We destroy the old currentGen so we can fill this empty array with all the elements of newGen
+        currentGen = [];
+
+        //Add current to all of our selected generations and push them into current Gen
+        for(var i=0; i < newGen.length; i++) {
+            $("#gen" + newGen[i]).addClass("current");
+            currentGen.push(newGen[i]);
+        }
+    } else if ($(".genSelect").hasClass('selected')) {
+        //In the case that the user began to change the generation and then changed their mind, 
+        //we switch the generations back to current
+        $('.genSelect.selected').toggleClass('selected current');
+
     }
     
+
     if(newDifficulty != currentDifficulty) {
         // The difficulty has been updated, so highlight the new one
-        document.getElementById('diff' + currentDifficulty).className = document.getElementById('diff' + currentDifficulty).className.replace('current','');
-        document.getElementById('diff' + newDifficulty).className = document.getElementById('diff' + currentDifficulty).className.replace('selected','');
-        document.getElementById('diff' + newDifficulty).className += ' current';
+
+        $(".diffSelect").removeClass("current selected");
+        $("#diff" + newDifficulty).addClass("current");
         
         // Show the info box explaining that the change means different streaks and times
-        document.getElementById('infoBoxRight').setAttribute('style', 'display: inherit');
-        
-        currentDifficulty = newDifficulty;      
+        $("#infoBoxRight").show();
+        currentDifficulty = newDifficulty;
+
     } else {
-        document.getElementById('infoBoxRight').setAttribute('style', 'display: none');
+        $("#infoBoxRight").hide();
     }
         
     // We're into a sound-based difficulty
     if(currentDifficulty > 2) {
-        document.getElementById('canvasContainer').setAttribute('style', 'display: none');
-        document.getElementById('pokemonCryPlayer').setAttribute('controls', 'controls');
+        $("#pokemonCryPlayer").show();
```

Comment:
Here and in a few other places and swapped out the older style changing for jquery style changing. I can continue this job if you think it's worth while to make the code more consistent